src/rpdk/core/contract/resource_generator.py

```diff
@@ -247,7 +256,7 @@ def generate_string_strategy(schema):
             if "maxLength" in schema:  # pragma: no cover
                 LOG.warning("found maxLength used with pattern")
 
-            return from_regex(terminate_regex(regex))
+            return from_regex(re.compile(terminate_regex(regex), re.ASCII))
```

Comment:
should terminate regex also account for the `\A`

Reply:
terminate_regex also set to replace `^` with `\A`